src/main/java/eu/bbmri_eric/negotiator/service/NetworkServiceImpl.java

```diff
@@ -0,0 +1,128 @@
+import eu.bbmri_eric.negotiator.database.repository.PersonRepository;
+import eu.bbmri_eric.negotiator.database.repository.ResourceRepository;
+import eu.bbmri_eric.negotiator.dto.network.NetworkCreateDTO;
+import eu.bbmri_eric.negotiator.dto.network.NetworkDTO;
+import eu.bbmri_eric.negotiator.exceptions.EntityNotFoundException;
+import eu.bbmri_eric.negotiator.exceptions.EntityNotStorableException;
+import eu.bbmri_eric.negotiator.exceptions.UserNotFoundException;
+import jakarta.transaction.Transactional;
+import java.util.List;
+import org.modelmapper.ModelMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@Transactional
+@Service
+public class NetworkServiceImpl implements NetworkService {
+  @Autowired NetworkRepository networkRepository;
+  @Autowired ResourceRepository resourceRepository;
+  @Autowired PersonRepository personRepository;
+  @Autowired ModelMapper modelMapper;
+
+  @Override
+  public NetworkDTO findNetworkById(Long id) {
+    return modelMapper.map(
+        networkRepository.findById(id).orElseThrow(() -> new EntityNotFoundException(id)),
+        NetworkDTO.class);
+  }
+
+  @Override
+  public Iterable<NetworkDTO> findAllNetworks(Pageable pageable) {
+    return networkRepository
+        .findAll(pageable)
+        .map(network -> modelMapper.map(network, NetworkDTO.class));
+  }
+
+  @Override
+  public void deleteNetworkById(Long id) {
+    Network network = getNetwork(id);
+    networkRepository.delete(network);
+  }
+
+  @Override
+  public void removeResourceFromNetwork(Long networkId, Long resourceId) {
+    Network network = getNetwork(networkId);
+    Resource resource = getResource(resourceId);
+    network.removeResource(resource);
+    networkRepository.save(network);
+  }
+
+  @Override
+  public void removeManagerFromNetwork(Long networkId, Long managerId) {
+    Network network = getNetwork(networkId);
+    Person manager = getManager(managerId);
+    network.removeManager(manager);
+    networkRepository.save(network);
+  }
+
+  @Override
+  public NetworkDTO updateNetwork(Long id, NetworkCreateDTO networkDTO)
+      throws EntityNotStorableException {
+    Network network = getNetwork(id);
+    network.setName(networkDTO.getName());
+    network.setUri(networkDTO.getUri());
+    network.setExternalId(networkDTO.getExternalId());
+    network.setContactEmail(networkDTO.getContactEmail());
+    networkRepository.save(network);
+    return modelMapper.map(network, NetworkDTO.class);
+  }
+
+  @Override
+  public NetworkDTO createNetwork(NetworkCreateDTO networkCreateDTO)
+      throws EntityNotStorableException {
+    Network network = modelMapper.map(networkCreateDTO, Network.class);
+    try {
+      Network savedNetwork = networkRepository.saveAndFlush(network);
+      return modelMapper.map(savedNetwork, NetworkDTO.class);
```

Comment:
this is technically not correct. The entity could have been saved but the mapper returns an error. I would rather mark this as transactional and just save it. What exceptions do you want to handle?